algorithms/C/sorting/counting-sort.c

```diff
@@ -0,0 +1,33 @@
+#include<stdio.h>
+ 
+void counting_sort(int a[],int n,int max)
+{
+     int count[50]={0},i,j;
+    
+     for(i=0;i<n;++i)
+      count[a[i]]=count[a[i]]+1;
+      
+     printf("\nSorted elements are: ");
+    
+     for(i=0;i<=max;++i)
+      for(j=1;j<=count[i];++j)
+       printf("%d ",i);
+}
+ 
+int main()
+{
+    int a[50],n,i,max=0;
+    printf("Enter number of elements: ");

```

Comment:
If the buffer length ( `n` in this case ) is meant to be taken interactively, then why initialise it with a fixed size before ? Wouldn't it cause problems involving insufficient space if the value of `n` is larger than the 50 or space wastage if it's less than that.

It's recommended to